aws/aws-parallelcluster
# [DFSM] Make validation of EBS volume id succeed when the volume is attached to the head node.

### Description of changes
1. Make validation of EBS volume id succeed when the volume is attached to the head node.
2. Prepared code to support multi-attach (commented out because we do not officially support it).
3. Added unit tests

**Note:** I think this validator should fail with an error when attempting to mount a not available volume without multi-attach, but I'd keep it as it was in previous releases at least in this PR because such a change required product approval. 

**Without this change**, whenever a user update a cluster with an EBS volume already mounted to it, the following error is emitted:

```
    {
      "level": "WARNING",
      "type": "SharedEbsVolumeIdValidator",
      "message": "Volume vol-0725fd369d04f4937 is in state 'in-use' not 'available'."
    },
```

The above is undesired because the volume is in-use because it is actually already mounted to the cluster.

### Tests
1. Create with available EBS volume => success
2. Create with in-use EBS volume => warning
4. Update with available EBS volume => success
5. Update with in-use EBS volume attached to the head node => success
6. Create/Update with in-use EBS volume having multi-attach => warning

Double-checked that there is no `_register_validators()` call without a context:
```
cat cli/src/**/*.py | grep _register_validators | wc -l          
      63
cat cli/src/**/*.py | grep _register_validators | grep context | wc -l
      63
cat cli/src/**/*.py | grep "def _register_validators" | wc -l                                    
      49
cat cli/src/**/*.py | grep "def _register_validators" | grep "context: ValidatorContext = None" | wc -l
      49

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Signed-off-by: Giacomo Marciani <mgiacomo@amazon.com>
